webhooks/virtualmachine/mutation/virtualmachine_mutator.go

```diff
@@ -450,3 +456,29 @@ func SetCreatedAtAnnotations(ctx context.Context, vm *vmopv1.VirtualMachine) {
 	vm.Annotations[constants.CreatedAtBuildVersionAnnotationKey] = pkgcfg.FromContext(ctx).BuildVersion
 	vm.Annotations[constants.CreatedAtSchemaVersionAnnotationKey] = vmopv1.SchemeGroupVersion.Version
 }
+
+// SetLastResizeAnnotation sets the last resize annotation as needed when the class name changes.
+func SetLastResizeAnnotation(
+	ctx *pkgctx.WebhookRequestContext,
+	vm, oldVM *vmopv1.VirtualMachine) (bool, error) {
+
+	if !pkgcfg.FromContext(ctx).Features.VMResize {
+		return false, nil
+	}
+
+	if vm.Spec.ClassName == oldVM.Spec.ClassName {
+		return false, nil
+	}
+
+	if _, _, _, ok := vmopv1util.GetLastResizedAnnotation(*vm); !ok {
```

Comment:
I know I mentioned some thoughts about removing the annotation if the class name changes back to what's saved in the annotation but I think in some situations that can result in some unexpected results so that's punted for now until I can think a bit more about it.